irohad/torii/command_service.hpp

```diff
@@ -46,33 +43,46 @@ namespace torii {
 
     CommandService(const CommandService &) = delete;
     CommandService &operator=(const CommandService &) = delete;
+
     /**
-     * actual implementation of async Torii in CommandService
+     * actual implementation of sync Torii in CommandService
      * @param request - Transaction
      * @param response - ToriiResponse
      */
     void Torii(iroha::protocol::Transaction const &request,
-                    google::protobuf::Empty &response);
-
-    void Status(iroha::protocol::TxStatusRequest const &request,
-                     iroha::protocol::ToriiResponse &response);
+               google::protobuf::Empty &response);
 
     virtual grpc::Status Torii(grpc::ServerContext *context,
                                const iroha::protocol::Transaction *request,
                                google::protobuf::Empty *response) override;
 
+    void Status(iroha::protocol::TxStatusRequest const &request,
+                iroha::protocol::ToriiResponse &response);
+
     virtual grpc::Status Status(
         grpc::ServerContext *context,
         const iroha::protocol::TxStatusRequest *request,
         iroha::protocol::ToriiResponse *response) override;
 
+    void StatusStream(
+        iroha::protocol::TxStatusRequest const &request,
+        grpc::ServerWriter<iroha::protocol::ToriiResponse> &response_writer);
+
+    virtual grpc::Status StatusStream(
+        grpc::ServerContext *context,
+        const iroha::protocol::TxStatusRequest *request,
+        grpc::ServerWriter<iroha::protocol::ToriiResponse> *response_writer)
+        override;
+
    private:
     std::shared_ptr<iroha::model::converters::PbTransactionFactory> pb_factory_;
     std::shared_ptr<iroha::torii::TransactionProcessor> tx_processor_;
     std::shared_ptr<iroha::ametsuchi::Storage> storage_;
     std::shared_ptr<
         iroha::cache::Cache<std::string, iroha::protocol::ToriiResponse>>
         cache_;
+    std::mutex wait_subscription_;
+    std::condition_variable cv_;
```

Comment:
Add documentation with reasons to use the conditional variable and write that data that guards under the cv.